gradio/blocks.py

```diff
@@ -489,13 +493,12 @@ def postprocess_data(self, fn_index, predictions, state):
             for i, output_id in enumerate(dependency["outputs"]):
                 block = self.blocks[output_id]
                 if getattr(block, "stateful", False):
-                    state[output_id] = predictions[i]
+                    if not is_update(predictions[i]):
```

Comment:
Why not just give `Variable()` an barebones `update()` method like other components instead of treating it as a special case?